doc/function.md

```diff
@@ -1,5 +1,282 @@
+
+### New
+
+Creates an instance of a `Function` object.
+
+```cpp
+template <typename Callable>
+static Function New(napi_env env, Callable cb, const char* utf8name = nullptr, void* data = nullptr);
+```
+
+- `[in] env`: The `napi_env` environment in which to construct the Function object.
+- `[in] cb`: Object that implements `Callable`.
+- `[in] utf8name`: Null-terminated string to be used as the name of the function.
+- `[in] data`: User-provided data context. This will be passed back into the
+function when invoked later.
+
+Returns an instance of a `Function` object.
+
+### New
+
+```cpp
+template <typename Callable>
+static Function New(napi_env env, Callable cb, const std::string& utf8name, void* data = nullptr);
+```
+
+- `[in] env`: The `napi_env` environment in which to construct the Function object.
+- `[in] cb`: Object that implements `Callable`.
+- `[in] utf8name`: String to be used as the name of the function.
+- `[in] data`: User-provided data context. This will be passed back into the
+function when invoked later.
+
+Returns an instance of a `Function` object.
+
+### New
+
+Creates a new JavaScript value from one that represents the constructor for the
+object.
+
+```cpp
+Object New(const std::initializer_list<napi_value>& args) const;
+```
+
+- `[in] args`: Initializer list of JavaScript values as `napi_value` representing
+the arguments of the contructor function.
+
+Returns a new JavaScript object.
+
+### New
+
+Creates a new JavaScript value from one that represents the constructor for the
+object.
+
+```cpp
+Object New(const std::vector<napi_value>& args) const;
+```
+
+- `[in] args`: Vector of JavaScript values as `napi_value` representing the
+arguments of the constructor function.
+
+Returns a new JavaScript object.
+
+### New
+
+Creates a new JavaScript value from one that represents the constructor for the
+object.
+
+```cpp
+Object New(size_t argc, const napi_value* args) const;
+```
+
+- `[in] argc`: The number of the arguments passed to the contructor function.
+- `[in] args`: Array of JavaScript values as `napi_value` representing the
+arguments of the constructor function.
+
+Returns a new JavaScript object.
+
+### Call
+
+Calls a Javascript function from a native add-on.
+
+```cpp
+Value Call(const std::initializer_list<napi_value>& args) const;
+```
+
+- `[in] args`: Initializer list of JavaScript values as `napi_value` representing
+the arguments of the function.
+
+Returns a Value representing the JavaScript object returned by the function.
+
+### Call
+
+Calls a Javascript function from a native add-on.
+
+```cpp
+Value Call(const std::vector<napi_value>& args) const;
+```
+
+- `[in] args`: Vector of JavaScript values as `napi_value` representing the
+arguments of the function.
+
+Returns a `Value` representing the JavaScript value returned by the function.
+
+### Call
+
+Calls a Javascript function from a native add-on.
+
+```cpp
+Value Call(size_t argc, const napi_value* args) const;
```

Comment:
Hmmm, come to think of it, should the return type not also be given as `Napi::Value`? I thought we had agreed to be explicit about the name space.